onnxruntime/core/providers/dnnl/dnnl_execution_provider.cc

```diff
@@ -58,6 +65,35 @@ DNNLExecutionProvider::DNNLExecutionProvider(const DNNLExecutionProviderInfo& in
   if (!fusion_env.empty()) {
     enable_fusion_ = (std::stoi(fusion_env) == 0 ? false : true);
   }
+
+  // Set the number of threads specified by the user
+  // If provided arguments set them as the number of threads, else call
+  // calc which usually = numcores
```

Comment:
What happens if environment variable omp_num_threads is set?

Reply:
According to the [documentation](https://learn.microsoft.com/en-us/cpp/parallel/openmp/reference/openmp-environment-variables?view=msvc-170): calling `set_omp_num_threads(num)` or `num_threads(num)` overwrites the environment variable `OMP_NUM_THREADS`.

Reply:
Can the number of threads be set only in onnxruntime_perf_test? How do you set number of threads if you're not using onnxruntime_perf_test?

Reply:
Of all the ORT tests perf test is the only one that provides user access to ep options so yes, this is the only test where you can set the number of threads.
Let's say we are creating our own app, when we create the oneDNN EP we have to pass the DnnlProviderOptions struct as an argument to create the EP, using that struct is how the user controls the number of threads.

Reply:
It's useful to set number of threads in scenarios besides onnxruntime_perf_test, e.g. debugging unit tests, onnx_test_runner. Is it possible to use the environment variable omp_num_threads in other scenarios?

Reply:
We could add the option to use the environment variable but to enable it the user would need to set a flag in the DnnlProviderOptions struct, if the user can do that then it might as well just use the struct to set the number of threads it wants.
The other way I can think of doing this would be to not fix the oversubscription issue and just let the user control the number of threads using the env variable.

Reply:
Nevermind, we can do it